### PR TITLE
Added datatypes transformations for Oracle TargetDB

### PR DESCRIPTION
**Pull request by priyanshi-yb**
Jenkins pipeline - 
1. Without debezium - https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/1013
2. debezium - https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/1014

### CODE REVIEW

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/schemaRegistry.go:44]: Todo for later: we should probably store this as a map instead of a list, so that it's faster to lookup column from columnName. I suppose this code is in the critical path.

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/schemaRegistry.go:47]: What about for CHAR, TIMESTAMP, LONG? You've also included those types int he debezium config.

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/suites/oracleSuite.go:140]: I think we got rid of this? Just be careful when merging

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/suites/oracleSuite.go:146]: uff, the interval handling looks like a lot of manually written logic 😅  
Have you checked interval.handling.mode in debezium? THere is a numeric option. Maybe that might be ingestible to oracle directly?

File: yb-voyager/src/tgtdb/suites/yugabytedbSuite.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/suites/yugabytedbSuite.go:1]: Assuming this is unchanged, you've just moved to a diff file.

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/oracle.go:638]: do we still need this given that we're getting the source data types from debezium.

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/oracle.go:594]: is 1-9 the precision of the milliseconds? Can you just add documentation in comments?

File: yb-voyager/src/dbzm/valueConverter.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/valueConverter.go:153]: Let's discuss this once on call? I want to understand the cyclic dependency once.

**Review by makalaaneesh — COMMENTED** [review]
Added some comments. Let me know after you address them & also merge main.

File: yb-voyager/src/dbzm/valueConverter.go
Comment (priyanshi-yb) [reply yb-voyager/src/dbzm/valueConverter.go:151]: yes sure

File: yb-voyager/src/tgtdb/oracle.go
Comment (priyanshi-yb) [reply yb-voyager/src/tgtdb/oracle.go:731]: yes, its precision, will add comment

File: yb-voyager/src/tgtdb/suites/yugabytedbSuite.go
Comment (priyanshi-yb) [reply yb-voyager/src/tgtdb/suites/yugabytedbSuite.go:1]: yes same

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (priyanshi-yb) [reply yb-voyager/src/tgtdb/suites/oracleSuite.go:146]: I can check that. Currently, the way debezium dumps interval data is easily ingested using COPY as is to YB but with SQlldr we need some transformation.

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (priyanshi-yb) [reply yb-voyager/src/tgtdb/suites/oracleSuite.go:140]: no, we are still formatting the values. It's just for prepared statements I am not formatting these values by unsettling the `formatIfRequired`.

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (priyanshi-yb) [reply yb-voyager/src/dbzm/schemaRegistry.go:47]: there are different usage of all of them. See below else-if conditions. for example for CHAR types I am using length to set in the CTL file.

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (priyanshi-yb) [reply yb-voyager/src/dbzm/schemaRegistry.go:44]: Yes agreed!

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (makalaaneesh) [reply yb-voyager/src/dbzm/schemaRegistry.go:47]: ok, just wanted to ensure you didn't miss it

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [reply yb-voyager/src/tgtdb/oracle.go:731]: oops my bad.

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (makalaaneesh) [reply yb-voyager/src/tgtdb/suites/oracleSuite.go:146]: Ah, right. forgot that we also need to worry about YB 😅 
Ok, just see if the numeric option makes it easy. Otherwise, if you've tested this logic, leave it for now, and we can revisit as this is not a very highly used data type?

Comment (priyanshi-yb) [reply yb-voyager/src/tgtdb/suites/oracleSuite.go:146]: I checked the numeric type it gives values in microseconds e.g. -
```
"1","-44706600000000","-1560000000"
```
 COPY can't work with this. I haven't tested sqlldr yet but I think it will also not be able to handle this and need to be manipulated to some understandable format. 
I think the current string format is fine. I have also tested my logic and it works fine.

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/suites/oracleSuite.go:61]: @priyanshi-yb pls take into consideration here the fixes i made in https://github.com/yugabyte/yb-voyager/pull/1023

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/schemaRegistry.go:43]: Leave it to you whether you want to do it now, or add a TODO for later:
it would be better to add a param that is at the same abstraction level as that of the schemaRegistry. 

so, instead of taking the targetDBType as input, how about you take a boolean flag `returnSourceDbType` which the caller will set to true. 

basically a similar API as that of `debezium.source.datatype.propagate.source.type`

Basically, the main concern being - SchemaRegistry and TableSchema should have no awareness of what the targetDBTYpe is

File: yb-voyager/src/dbzm/schemaRegistry.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/schemaRegistry.go:77]: consider setting targetDBType at the schemaRegistry level, so you won't have to change these   APIs.

File: yb-voyager/src/dbzm/valueConverter.go
Comment (makalaaneesh) [reply yb-voyager/src/dbzm/valueConverter.go:172]: pls add a TODO here to fix

File: yb-voyager/src/tgtdb/target_db_interface.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/target_db_interface.go:127]: Add a TODO comment here as well. 
It seems mixed up for the importBatch layer to have awareness of debezium. Given that we absolutely need it - we should call it something else - like `source_db_datatype`

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (makalaaneesh) [reply yb-voyager/src/tgtdb/suites/oracleSuite.go:61]: Also, test timestamp fields while streaming.

File: yb-voyager/cmd/importData.go
Comment (makalaaneesh) [reply yb-voyager/cmd/importData.go:188]: I'm somehow still not clear on the case sensitivity handling here. I thought we didn't have to deal with quotedStrings? Let's discuss this once.

File: yb-voyager/src/dbzm/valueConverter.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/valueConverter.go:186]: (again for the future :D ) A probably nice way to deal with this might be: 
- instead of returning just a single columnType string for each column, return a column struct (that has all the information pertaining to that column - the name, the source type, the debezium type). 

Then,
- the valueConverter can work with that field struct (as opposed to a single keyword right now). 
- we can embed each column's schema in the data file descriptor (similar to how we add the delimiter/quote char/etc info in it. Then, we won't have to pass the schema directly to importBatch. 

Makes sense? Just add a TODO with some notes on how we can make this better later.

File: yb-voyager/src/dbzm/valueConverter.go
Comment (makalaaneesh) [inline yb-voyager/src/dbzm/valueConverter.go:189]: Let's remember to clean up this dependency mess immediately post the release.

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/oracle.go:47]: Doesn't seem like you're using this?

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/oracle.go:656]: Let's discuss this on call :) I'll need context

File: yb-voyager/src/tgtdb/suites/oracleSuite.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/suites/oracleSuite.go:103]: Hmm, I'm a little unsure about this. What is this scenario where we don't have to include timezone information while inserting? It's a TZ included field, so I think we should always make the timezone explicit.

File: yb-voyager/src/tgtdb/target_db_interface.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/target_db_interface.go:127]: how come this field is always present? I would have thought you should check for existence first?

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [reply yb-voyager/src/tgtdb/oracle.go:656]: to discuss: are we making a call to TgtDB in the critical path?

File: yb-voyager/src/tgtdb/oracle.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/oracle.go:655]: Important: test streaming phase for these column types.

File: yb-voyager/src/tgtdb/suites/yugabytedbSuite.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/suites/yugabytedbSuite.go:98]: change to UTC()

File: yb-voyager/src/tgtdb/target_db_interface.go
Comment (makalaaneesh) [inline yb-voyager/src/tgtdb/target_db_interface.go:129]: let's use a switch case instead here.

**Review by makalaaneesh — APPROVED** [review]
LGTM

**Pull request merged**
